src/ex2/ItemManager.js

```diff
@@ -0,0 +1,179 @@
+      if (this.pokemons.has(`${pokemon.name}`))
+        alert(`Pokemon with ID ${pokemon.id} was alrady added!`); //alert existing pokemons
+      return !this.pokemons.has(`${pokemon.name}`);
+    });
+
+    // add new pokemons to list of pokemons
+    const pokemonNames = newPokemons.map((pokemon) => {
+      if (pokemon.name) {
+        this.pokemons.add(`${pokemon.name}`);
+        return pokemon.name;
+      } else {
+        return pokemon;
+      }
+    });
+
+    // push pokemons to todo list
+    pokemonNames.forEach((name) => {
+      const id = name.split(" ")[0];
+
+      if (!isNaN(+id)) {
+        //if first element is a number - it's id of a not found pokemon
+        this.itemList.push(`Pokemon with ID ${id} was not found`);
+      } else {
+        //if first element is not a number - it's name of a found pokemon
+        this.itemList.push(`Catch ${name}`);
+      }
+    });
+  }
+
+  createPromises(inputArray) {
+    let allPromises = [];
+
+    inputArray.forEach((elm) => {
+      const id = elm.trim();
+
+      allPromises.push(pokemonClient.fetchPokemon(id));
+    });
+
+    return allPromises;
+  }
+
+  removeItem(e, liElm) {
+    e.stopPropagation();
+
+    const itemId = liElm.id.split("-");
+    const pokemonName = itemId[1];
+    const item = itemId.join(" ");
+
+    const itemIndex = this.itemList.findIndex((listItem) => {
+      return listItem === item;
+    });
+
+    this.itemList.splice(itemIndex, 1);
+    this.pokemons.delete(pokemonName);
+
+    this.renderItems();
+  }
+
+  removeAll() {
+    this.itemList.length = 0;
+    this.pokemons.clear();
+
+    this.renderItems();
+  }
+
+  renderItems(current) {
+    this.toggleFooter();
+
+    // clear list innerHTML
+    const list = document.querySelector("#list");
+    list.innerHTML = "";
+
+    // create elements for exisitng items
+    this.itemList.forEach((item) => {
+      const itemNode = this.createItemElement(item, current);
+
+      list.appendChild(itemNode);
+    });
+  }
+
+  toggleFooter() {
+    const addItemButton = document.querySelector("#list-item-submit");
+    const clearAllButton = document.querySelector("#clear-all-button");
+    const sortByNameButton = document.querySelector("#sort-by-name-button");
+
+    if (this.itemList.length === 0) {
+      addItemButton.classList.add("hithere"); //add AddButton animation
+      clearAllButton.classList.add("hidden"); //hide ClearAll button
+      sortByNameButton.classList.add("hidden"); //hide Sort button
+    } else {
+      addItemButton.classList.remove("hithere"); //remove AddButton animation
+      clearAllButton.classList.remove("hidden"); //show ClearAll button
+      sortByNameButton.classList.remove("hidden"); //show Sort button
+    }
+  }
+
+  createItemElement(input, current) {
+    const itemId = input.split(" ").join("-");
+
+    //create list element, add class, innerHTML, eventListener
+    const liElm = document.createElement("li");
```

Comment:
all this would be better in a single responsibility internal function that could create the listElement, making this function cleaner and more readable